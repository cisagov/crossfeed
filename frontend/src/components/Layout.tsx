import React from 'react';
import { useLocation } from 'react-router-dom';
import ScopedCssBaseline from '@material-ui/core/ScopedCssBaseline';
import { makeStyles } from '@material-ui/core/styles';
import { Header, GovBanner } from 'components';

const useStyles = makeStyles((theme) => ({
  root: {
    position: 'relative',
    height: '100vh',
    display: 'flex',
    flexFlow: 'column nowrap',
    overflow: 'auto'
  },
  overrides: {
    WebkitFontSmoothing: 'unset',
    MozOsxFontSmoothing: 'unset'
  },
  content: {
    flex: '1',
    display: 'block',
    position: 'relative'
  }
}));

export const Layout: React.FC = ({ children }) => {
  const classes = useStyles();
  const { pathname } = useLocation();

  return (
    <ScopedCssBaseline classes={{ root: classes.overrides }}>
      <div className={classes.root}>
        <GovBanner />
        <Header />

        {pathname === '/inventory' ? (
          children
        ) : (
          <div className={classes.content}>{children}</div>
        )}
      </div>
    </ScopedCssBaseline>
  );
};
