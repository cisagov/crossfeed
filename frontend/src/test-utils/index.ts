export * from './test-utils';
export * from './user';
export * from './organization';
