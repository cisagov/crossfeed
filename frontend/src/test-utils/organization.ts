import { Organization } from 'types';

export const testOrganization = {
  id: 'd7d6e913-0370-4f43-aebc-6bd727adc70e',
  name: 'test org',
  rootDomains: ['example.com'],
  ipBlocks: [],
  userRoles: [],
  scanTasks: [],
  isPassive: false,
  inviteOnly: false,
  granularScans: []
};
