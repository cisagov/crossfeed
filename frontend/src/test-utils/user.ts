import { AuthUser } from 'context';

export const testUser: AuthUser = {
  id: 'd7d6e913-0370-4f43-aebc-6bd727adc70e',
  createdAt: '2020-08-23T03:36:57.231Z',
  updatedAt: '2020-08-23T03:36:57.231Z',
  firstName: 'John',
  lastName: 'Smith',
  fullName: 'John Smith',
  invitePending: false,
  userType: 'standard',
  email: 'test@crossfeed.gov',
  roles: [],
  dateAcceptedTerms: new Date(),
  acceptedTermsVersion: 'v1-user',
  isRegistered: true
};
