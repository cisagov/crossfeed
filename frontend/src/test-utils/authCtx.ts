import { AuthContextType } from 'context';
import { testUser } from './user';

export const authCtx: AuthContextType = {
  user: testUser,
  token: 'some-test-token-never-verified',
  currentOrganization: null,
  loading: false,
  login: jest.fn(),
  logout: jest.fn(),
  setUser: jest.fn(),
  setOrganization: jest.fn(),
  refreshUser: jest.fn(),
  setLoading: jest.fn(),
  apiGet: jest.fn(),
  apiDelete: jest.fn(),
  apiPatch: jest.fn(),
  apiPost: jest.fn(),
  apiPut: jest.fn(),
  maximumRole: 'user',
  touVersion: 'v1-user',
  userMustSign: false
};
