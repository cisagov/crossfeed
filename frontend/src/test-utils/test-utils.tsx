import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { mocked } from 'ts-jest/utils';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext, AuthContextType } from '../context/AuthContext';
import { CFThemeProvider } from 'context';
import { authCtx } from './authCtx';

interface CustomRenderOptions extends RenderOptions {
  initialHistory?: string[];
  authContext?: Partial<AuthContextType>;
}

const customRender = (ui: any, options: CustomRenderOptions = {}) => {
  const { initialHistory, authContext, ...rest } = options;

  // Provide any context that the components may be expecting
  const Wrapper: React.FC = ({ children }) => (
    <CFThemeProvider>
      <MemoryRouter initialEntries={initialHistory}>
        <AuthContext.Provider
          value={{
            ...authCtx,
            ...authContext
          }}
        >
          {children}
        </AuthContext.Provider>
      </MemoryRouter>
    </CFThemeProvider>
  );
  return render(ui, { wrapper: Wrapper, ...rest });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, mocked };
