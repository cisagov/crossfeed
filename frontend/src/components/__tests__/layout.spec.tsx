import React from 'react';
import { render } from 'test-utils/test-utils';
import { Layout } from '../Layout';

jest.mock('components/Header', () => ({
  Header: () => <div>HEADER</div>
}));
jest.mock('components/GovBanner', () => ({
  GovBanner: () => <div>GOV_BANNER</div>
}));

afterAll(() => {
  jest.restoreAllMocks();
});

describe('Layout component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Layout />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders children', () => {
    const { getByText } = render(<Layout>some children</Layout>);
    expect(getByText('some children')).toBeInTheDocument();
  });
});
