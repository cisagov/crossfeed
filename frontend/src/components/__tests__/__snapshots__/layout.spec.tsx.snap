// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiScopedCssBaseline-root makeStyles-overrides-3"
  >
    <div
      class="makeStyles-root-1"
    >
      <div>
        GOV_BANNER
      </div>
      <div>
        HEADER
      </div>
      <div
        class="makeStyles-content-2"
      />
    </div>
  </div>
</DocumentFragment>
`;
